src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/CannotSealTypeTests.cs

```diff
@@ -0,0 +1,381 @@
+            {
+                @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+    }
+}",
+                @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+        protected UnsealedClass() { }
+    }
+}",
+                @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+        internal UnsealedClass() { }
+    }
+}"
+            };
+
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            MetadataInformation leftMetadata = new("left", "net6.0", "ref/a.dll");
+            ElementContainer<IAssemblySymbol> leftContainer = new(left, leftMetadata);
+
+            IList<ElementContainer<IAssemblySymbol>> right = SymbolFactory.GetElementContainersFromSyntaxes(rightSyntaxes);
+
+            ApiComparer differ = new();
+            differ.IncludeInternalSymbols = true;
+            IEnumerable<(MetadataInformation left, MetadataInformation right, IEnumerable<CompatDifference> differences)> result =
+                differ.GetDifferences(leftContainer, right);
+
+            AssertExtensions.MultiRightEmptyDifferences(leftMetadata, 3, result);
+        }
+
+        [Theory]
+        [MemberData(nameof(StrictModeSealedLeftIsReportedData))]
+        public void StrictModeSealedLeftIsReported(string leftSyntax, string rightSyntax)
+        {
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+
+            ApiComparer differ = new();
+            differ.StrictMode = true;
+            CompatDifference[] differences = differ.GetDifferences(left, right).ToArray();
+
+            CompatDifference[] expected = new[]
+            {
+                new CompatDifference(DiagnosticIds.CannotSealType, string.Empty, DifferenceType.Changed, "T:CompatTests.UnsealedClass")
+            };
+
+            Assert.Equal(expected, differences, CompatDifferenceComparer.Default);
+            Assert.True(differences[0].Message.IndexOf("left") < differences[0].Message.IndexOf("right"));
+        }
+
+        public static IEnumerable<object[]> StrictModeSealedLeftIsReportedData()
+        {
+            yield return new[]
+            {
+                @"
+namespace CompatTests
+{
+    public sealed class UnsealedClass
+    {
+        public UnsealedClass() { }
+    }
+}
+",
+                @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+      public UnsealedClass() { }
+    }
+}
+"
+            };
+
+            yield return new[]
+            {
+                @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+        private UnsealedClass() { }
+    }
+}
+",
+                @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+        public UnsealedClass() { }
+    }
+}
+"
+            };
+        }
+
+        public static IEnumerable<object[]> SealInheritableTypeReportedData()
```

Comment:
can we add some static ctor tests as well ?

Reply:
I believe static ctor doesn't affect inherability of a type.

Reply:
Ah I see what you mean. Will add one.